components/utils/charts.ts

```diff
@@ -0,0 +1,134 @@
+// It's recommended to prefer only *increases* to saturation, since low saturation looks dull.
+export const getColorVariants = (
+  color: colorRgbHsl,
+  numberOfVariants: number,
+  LFactorRange: { from: number, to: number } = { from: 0.75, to: 1.25 },
+  SFactorRange: { from: number, to: number } = { from: 1, to: 1.25 }, // NB only one color does not already have maximum saturation.
+  opacity: number = 1,
+): string[] =>
+  Array.from({ length: numberOfVariants }, (_, i) => {
+    // Get the next equidistant factor in the range
+    const LFactor = LFactorRange.from + (LFactorRange.to - LFactorRange.from) * (i / (numberOfVariants - 1));
+    const originalL = color.hsl[2];
+    const newLightness = Math.max(0, Math.min(100, originalL * LFactor));
+
+    const SFactor = SFactorRange.from + (SFactorRange.to - SFactorRange.from) * (i / (numberOfVariants - 1));
+    const originalS = color.hsl[1];
+    const newSaturation = Math.max(0, Math.min(100, originalS * SFactor));
+
+    const newHsl: HSL = [color.hsl[0], newSaturation, newLightness];
+    const rgb = convert.hsl.rgb(newHsl);
+    return `rgba(${rgb[0]},${rgb[1]},${rgb[2]},${opacity})`;
+  }).reverse();
+
+export interface LegendItem {
+  color: string
+  label: string
+  shape: string
+}
+
+export enum LegendShape {
+  Rectangle = "rectangle",
+  Line = "line",
+  Circle = "circle",
+}
+
+export const contextButtonOptions = {
+  height: 20,
+  width: 22,
+  symbolSize: 12,
+  symbolY: 10,
+  symbolX: 12,
+  symbolStrokeWidth: 2,
+  // Omit 'printChart' and 'viewData' from menu items
+  menuItems: ["downloadCSV", "downloadXLS", "separator", "downloadPNG", "downloadJPEG", "downloadPDF", "downloadSVG", "separator", "viewFullscreen"],
+  useHTML: true,
+} as Highcharts.ExportingButtonsOptionsObject;
+
+export const menuItemDefinitionOptions = {
+  downloadCSV: {
+    text: "Download CSV for this chart",
+  },
+  downloadXLS: {
+    text: "Download XLS for this chart",
+  },
+} as Highcharts.Dictionary<Highcharts.ExportingMenuObject>;
+
+export const chartBackgroundColorOnExporting = "white";
+const chartBackgroundColor = "transparent";
+export const chartOptions = {
+  backgroundColor: chartBackgroundColor,
+  events: {
+    fullscreenOpen() {
+      this.update({ chart: { backgroundColor: chartBackgroundColorOnExporting } });
+    },
+    fullscreenClose() {
+      this.update({ chart: { backgroundColor: chartBackgroundColor } });
+    },
+  },
+  style: {
+    fontFamily: "ImperialSansText, sans-serif",
+  },
+} as Highcharts.ChartOptions;
+
+// The TS declarations for the context object aren't correct, so we declare our own interface:
+// https://github.com/highcharts/highcharts/issues/22281#issuecomment-2516994341
+// The interface appears to be the same as the now defunct Highcharts.TooltipFormatterContextObject class.
+export interface TooltipPointInstance extends Highcharts.Point {
+  points?: Array<TooltipPointInstance>
+  point?: TooltipPointInstance
+  total: number
+  x: number
```

Comment:
This line `x: number` is the only new line in this file, which was renamed from `highCharts.ts` to `charts.ts`.